### ISSUE TITLE
Possible to add several Bonds to the same Dynamic 

### ISSUE THREAD

**Issue by ivanmoskalev** [SwiftBond/Bond#78]
Wrote this little test and noticed it fails:

```swift
func testMultipleBonds() {
    let dynamicInt = Dynamic<Int>(0)

    var counter = 0
    let intBond = Bond<Int>({ value in
      counter++
    })

    dynamicInt.bonds.append(BondBox<Int>(intBond))
    dynamicInt.bonds.append(BondBox<Int>(intBond))
    dynamicInt.bonds.append(BondBox<Int>(intBond))

    dynamicInt.value = 1

    XCTAssert(counter == 1, "Bond called more than once")
}
```

Is this the intended behavior? Because I thought that one each `Dynamic -> Bond` binding is unique.

**Comment by srdanrasic** [comment]
This is incorrect way to bind. You should use `->>` operator or `bindTo` method.

It is fine to bind one Dynamic to multiple Bonds.

**Comment by ivanmoskalev** [comment]
I'm sorry that I didn't make clear what I intended to communicate in the initial post.

```swift
  func testMultipleBonds() {
    let dynamicInt = Dynamic<Int>(0)

    var counter = 0
    let intBond = Bond<Int>({ value in
      counter++
    })

    dynamicInt ->| intBond
    dynamicInt ->| intBond
    dynamicInt ->| intBond

    dynamicInt.value = 1

    XCTAssert(counter == 1, "Bond called more than once")
  }
```

As you may see, I'm binding the same `Bond` to the same `Dynamic`, three times. I use the `->|` operator which doesn't fire on binding. After that I change the value of a dynamic, and the bond's handler closure gets called three times.

I would expect that it would be called only once.

It happens because `bonds` property of `Dynamic` is an array, and not a set.
```
public var bonds: [BondBox<T>] = []
```

If this is the expected behavior, then let's close this issue! :) 

Otherwise, we make every `Bond` and `Dynamic` (and their `*Box`es) Hashable and Equatable (I guess, it is possible to use `UnsafePointer`s to achieve this), and then use `Set`s for referencing bindings.

**Comment by srdanrasic** [comment]
Ok, now I see :) Hmm, I guess I would not expect that behaviour. Set would be preferred in that case.

Thanks for bringing this up.

**Comment by ivanmoskalev** [comment]
I can submit a pull request if you like! :)

**Comment by srdanrasic** [comment]
That would be great :)

**Comment by srdanrasic** [comment]
I've released v3.8 with your improvements. Thanks a lot!

**Comment by ivanmoskalev** [comment]
I'm glad my little effort helped your great library! Keep up the awesome work! :+1:
